services/src/test/java/io/scalecube/services/ServiceRegistryImplTest.java

```diff
@@ -0,0 +1,49 @@
+package io.scalecube.services;
+
+import static org.junit.Assert.assertEquals;
+
+import io.scalecube.cluster.Cluster;
+import io.scalecube.services.annotations.AnnotationServiceProcessor;
+import io.scalecube.services.annotations.ServiceProcessor;
+import io.scalecube.transport.Transport;
+
+import org.junit.Test;
+
+public class ServiceRegistryImplTest {
+
+  @Test
+  public void test_service_registry(){
+    
+    Cluster cluster = Cluster.joinAwait();
+    Transport transport = Transport.bindAwait();
+
+    ServicesConfig services = ServicesConfig.empty();
+    ServiceProcessor serviceProcessor = new AnnotationServiceProcessor();
+    
+    ServiceRegistryImpl registry = new ServiceRegistryImpl(cluster, transport, services, serviceProcessor);
+   
+    try{
+     new ServiceRegistryImpl(null, transport, services, serviceProcessor);
+    } catch (Exception ex){
+      assertEquals(ex.toString(), "java.lang.IllegalArgumentException: cluster can't be null");
+    }
+    
+    try{
+      new ServiceRegistryImpl(cluster, null, services, serviceProcessor);
+     } catch (Exception ex){
+       assertEquals(ex.toString(), "java.lang.IllegalArgumentException: transport can't be null");
```

Comment:
Also it is better to have several separate tests for each of this case, at least it seems for me a bit cleaner.